cmd/manager/operator.go

```diff
@@ -221,31 +241,16 @@ func addMetrics(ctx context.Context, cfg *rest.Config) {
 		log.Info("Could not generate and serve custom resource metrics", "error", err.Error())
 	}
 
-	// Add to the below struct any other metrics ports you want to expose.
-	servicePorts := []v1.ServicePort{
-		{Port: metricsPort, Name: metrics.OperatorPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: metricsPort}},
-		{Port: operatorMetricsPort, Name: metrics.CRPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: operatorMetricsPort}},
-	}
-
-	// Create Service object to expose the metrics port(s).
-	service, err := metrics.CreateMetricsService(ctx, cfg, servicePorts)
+	// Fetch the metrics service created during the deployment.
+	metricsService, err := kClient.CoreV1().Services(operatorNs).Get(ctx, metricsServiceName, metav1.GetOptions{})
 	if err != nil {
-		log.Info("Could not create metrics Service", "error", err.Error())
+		log.Error(err, "Error getting metrics service")
+		os.Exit(1)
```

Comment:
Does this need to be a hard error? (Both yes and no are fine; see my other comment about that error being unchecked when starting up the metrics controller..the only reason I'm commenting is that I want to understand which piece we consider critical and which optional)

Reply:
I'm on the fence about all metrics pieces being a hard requirement or not. On one hand, it will be annoying to the user to be put in the spot where they have to troubleshoot why metrics aren't showing, even though the operator is running. If we crash it is more obvious..

Reply:
CO is (at least at the moment) an OCP-only operator and frankly I don't see it changing anytime soon. Personally I'm more in favor of requiring the critical functionality like metrics, provided the deps are there by default and just crash if something goes sideways.
We can always relax the requirement later if we get bug reports..

Reply:
good point, i'll fix